### ISSUE TITLE
Training loss diagnostics plot

### ISSUE THREAD

**Issue by radka-j** [alan-turing-institute/autoemulate#926]
Add `ae.plot_loss()` method for plotting training loss over time.

**Comment by radka-j** [comment]
That's great @TheCSGuy25!

Some notes on this:
- The method should work similarly to the other plotting functions (i.e., take in a `model_obj` that is a model/result object to plot the loss of)
- The `PytorchBackend` stores `loss_history` as an attribute so any emulator that inherits from this class should have this accessible - simply plot this over iterations
- The `GaussianProcess` emulator does not have a `loss_history` attribute but it should - this needs to be added (just follow what is done in the `PytorchBackend`)
- There are emulators that are not trained with gradient descent (particularly anything that inherits from `SklearnBackend` and `RadialBasisFunctions`) so the plotting function has to raise an error if one of these models is passed to it

**Comment by TheCSGuy25** [comment]
Thanks, will keep that in mind! Don't mind me when i reach out with questions.
